.github/workflows/lint.yml

```diff
@@ -0,0 +1,37 @@
+name: Lint
+
+on: [pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: ruby/setup-ruby@v1
+
+      - name: Get base and head
+        id: base-head
+        run: |
+          git fetch origin ${{ github.base_ref }} ${{ github.event.pull_request.head.ref }}
```

Comment:
https://github.com/orgs/community/discussions/25950#discussioncomment-8236983

This is a workaround for inconsistencies in the Github actions context.

Ideally we should be linting the entire repo, but for now I intend on getting parity with what we have in `v2-spec-helpers` gem.